pkg/repository/buffer/buffered.go

```diff
@@ -313,6 +405,14 @@ func (b *IngestBuf[T, U]) BuffItem(item T) (chan *FlushResponse[U], error) {
 		return nil, fmt.Errorf("buffer not ready, in state '%v'", b.state.String())
 	}
 
+	if b.safeCheckSizeOfBuffer() >= b.maxCapacity*50 {
+		return nil, status.Errorf(codes.ResourceExhausted, "buffer is out of space %v", b.safeCheckSizeOfBuffer())
```

Comment:
Why are we using GRPC status errors here?

Reply:
trying to make it work with the retry when we push that.